pkg/endpointmanager/policymap_pressure.go

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package endpointmanager
+
+import (
+	"sync/atomic"
+	"time"
+
+	"github.com/cilium/cilium/pkg/endpoint"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/maps/policymap"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/trigger"
+)
+
+func (p *policyMapPressure) Update(ev endpoint.PolicyMapPressureEvent) {
+	val := ev.Value
+
+	log.WithField(logfields.Value, val).Debug("EndpointManager policymap received event")
+
+	cur := p.current.Load()
+	if cur == nil || val > *cur {
+		p.current.Store(&val)
+		p.trigger.Trigger()
+	}
+}
+
+func newPolicyMapPressure() *policyMapPressure {
+	if !metrics.BPFMapPressure {
+		return nil
+	}
+
+	p := new(policyMapPressure)
+	p.gauge = metrics.NewBPFMapPressureGauge(policymap.MapName+"*", policymap.PressureMetricThreshold)
+
+	var err error
+	p.trigger, err = trigger.NewTrigger(trigger.Parameters{
+		// It seems like 10s is a small enough window of time where the user
+		// can still reasonably react to a rising BPF map pressure. Keep it
+		// below the default Prometheus scrape interval of 15s anyway.
+		MinInterval: 10 * time.Second,
+		TriggerFunc: func([]string) { p.update() },
+		Name:        "endpointmanager-policymap-max-size-metrics",
+	})
+	if err != nil {
+		log.WithError(err).Panic("Failed to initialize trigger for policymap pressure metric")
```

Comment:
It's not common to panic in the agent. Why is this an exception?

Reply:
In this case, the only way this panic gets triggered is if the `TriggerFunc` is nil. Given this code will initialize very early on in the daemon, it acts as a sanity check before proceeding. It is assumed that if this code panics, then it was a developer mistake. There is no risk that this panics in production because it is only called once on dameon init and it will be caught immediately in CI.

Reply:
I feel like we could probably drop the error condition from `NewTrigger()` and just force the caller to pass a function directly in, to ensure it isn't nil. It doesn't really make sense for a caller to initialize a trigger without a function to trigger, since that defeats the whole point of the trigger in the first place. That could help to avoid the need for this sort of awkward error checking + panic logic in arbitrary packages like this.

(I buy your argument in this specific scenario, though I am a bit paranoid in general about `Panic()`, specifically because of the potential for unexpected agent failure)